### DISCUSSION TITLE
Ability to set lightswitch value to an environment variable

### DISCUSSION THREAD

**Discussion by Mosnar** [craftcms/cms#8866] [Ideas] [answered]
### Description

It would be great if there was some way to easily include a lightswitch input that has the ability to be set to a boolean-like environment variable. Admittedly, I'm having a hard time imagining what this UI would look like, but I think it's an important feature when project config makes plugin settings inaccessible. The most obvious use-case is for "dev mode" type lightswitches.

**Comment by brandonkelly** [comment]
I’m not sure I’m following… can you give me a more concrete example?

**Comment by Mosnar** [comment]
Sure - here are a couple of examples:

From the AvaTax plugin (note, using a config file to override these settings, ultimately achieving what I'm looking for)
![image](https://user-images.githubusercontent.com/3810939/87837850-01753f00-c863-11ea-9417-cd6674be0a60.png)

From a Commerce Gateway:
![image](https://user-images.githubusercontent.com/3810939/87837918-426d5380-c863-11ea-8768-1862570e135b.png)

In non-production environments, the gateway uses keys for a sandbox environment, but it only works if developer mode is enabled via the lightswitch.

The work-around is to do what the AvaTax plugin does, which is allow for overriding via a config file. That feels a little inconsistent/antiquated to me, since we have the ability to enter environment variables all over the control panel now.

**Comment by nettum** [comment]
This could also be useful for other fieldtypes.
E.g. if you want different email settings between envs in craft and using project config / allowAdminChanges.
All text fields can be set to env variables in the UI, but not the checkbox (useAuthentication) an the dropdowns (transport and encryption).

**Comment by brandonkelly** [comment]
Noticed Zapier has a pretty elegant solution for this sort of thing. In their case you might want to pull in a boolean value from a prior step in the zap, which is sortof like needing to pull in an environment variable.

<img width="841" alt="Screenshot of a boolean “Text to Speech” setting within a Zapier zap, with a dropdown menu expanded below it with “True” and “False” radio buttons, plus a “Custom” tab" src="https://user-images.githubusercontent.com/47792/123124670-b8c3ef80-d3fc-11eb-9aa9-d134f089400f.png">

<img width="830" alt="The “Custom” tab within the dropdown, showing that you can select various values from prior steps in the zap" src="https://user-images.githubusercontent.com/47792/123124983-fc1e5e00-d3fc-11eb-894c-654a535af374.png">

We’re going to get this into v4.

**Comment by brandonkelly** [comment, marked as answer]
Just resolved this a little ahead of schedule, for the next Craft 3 release 🚀 

See details at #10111

**Reply by brandonkelly** [reply]
Craft 3.7.22 is out now with those changes.
